meson.build

```diff
@@ -0,0 +1,717 @@
+    'scintilla/src/Indicator.cxx',
+    'scintilla/src/Indicator.h',
+    'scintilla/src/IntegerRectangle.h',
+    'scintilla/src/KeyMap.cxx',
+    'scintilla/src/KeyMap.h',
+    'scintilla/src/LineMarker.cxx',
+    'scintilla/src/LineMarker.h',
+    'scintilla/src/MarginView.cxx',
+    'scintilla/src/MarginView.h',
+    'scintilla/src/Partitioning.h',
+    'scintilla/src/PerLine.cxx',
+    'scintilla/src/PerLine.h',
+    'scintilla/src/Position.h',
+    'scintilla/src/PositionCache.cxx',
+    'scintilla/src/PositionCache.h',
+    'scintilla/src/RESearch.cxx',
+    'scintilla/src/RESearch.h',
+    'scintilla/src/RunStyles.cxx',
+    'scintilla/src/RunStyles.h',
+    'scintilla/src/SVector.h',
+    'scintilla/src/ScintillaBase.cxx',
+    'scintilla/src/ScintillaBase.h',
+    'scintilla/src/Selection.cxx',
+    'scintilla/src/Selection.h',
+    'scintilla/src/SparseVector.h',
+    'scintilla/src/SplitVector.h',
+    'scintilla/src/Style.cxx',
+    'scintilla/src/Style.h',
+    'scintilla/src/UniConversion.cxx',
+    'scintilla/src/UniConversion.h',
+    'scintilla/src/UnicodeFromUTF8.h',
+    'scintilla/src/UniqueString.cxx',
+    'scintilla/src/UniqueString.h',
+    'scintilla/src/ViewStyle.cxx',
+    'scintilla/src/ViewStyle.h',
+    'scintilla/src/XPM.cxx',
+    'scintilla/src/XPM.h',
+    'scintilla/lexers/LexAbaqus.cxx',
+    'scintilla/lexers/LexAda.cxx',
+    'scintilla/lexers/LexAsm.cxx',
+    'scintilla/lexers/LexBash.cxx',
+    'scintilla/lexers/LexBasic.cxx',
+    'scintilla/lexers/LexBatch.cxx',
+    'scintilla/lexers/LexCOBOL.cxx',
+    'scintilla/lexers/LexCPP.cxx',
+    'scintilla/lexers/LexCSS.cxx',
+    'scintilla/lexers/LexCaml.cxx',
+    'scintilla/lexers/LexCmake.cxx',
+    'scintilla/lexers/LexCoffeeScript.cxx',
+    'scintilla/lexers/LexD.cxx',
+    'scintilla/lexers/LexDiff.cxx',
+    'scintilla/lexers/LexErlang.cxx',
+    'scintilla/lexers/LexForth.cxx',
+    'scintilla/lexers/LexFortran.cxx',
+    'scintilla/lexers/LexHTML.cxx',
+    'scintilla/lexers/LexHaskell.cxx',
+    'scintilla/lexers/LexJulia.cxx',
+    'scintilla/lexers/LexLaTeX.cxx',
+    'scintilla/lexers/LexLisp.cxx',
+    'scintilla/lexers/LexLua.cxx',
+    'scintilla/lexers/LexMake.cxx',
+    'scintilla/lexers/LexMarkdown.cxx',
+    'scintilla/lexers/LexMatlab.cxx',
+    'scintilla/lexers/LexNsis.cxx',
+    'scintilla/lexers/LexNull.cxx',
+    'scintilla/lexers/LexPascal.cxx',
+    'scintilla/lexers/LexPerl.cxx',
+    'scintilla/lexers/LexPowerShell.cxx',
+    'scintilla/lexers/LexProps.cxx',
+    'scintilla/lexers/LexPython.cxx',
+    'scintilla/lexers/LexPO.cxx',
+    'scintilla/lexers/LexR.cxx',
+    'scintilla/lexers/LexRuby.cxx',
+    'scintilla/lexers/LexRust.cxx',
+    'scintilla/lexers/LexSmalltalk.cxx',
+    'scintilla/lexers/LexSQL.cxx',
+    'scintilla/lexers/LexTCL.cxx',
+    'scintilla/lexers/LexTxt2tags.cxx',
+    'scintilla/lexers/LexVHDL.cxx',
+    'scintilla/lexers/LexVerilog.cxx',
+    'scintilla/lexers/LexYAML.cxx',
+    cpp_std: 'c++17',
```

Comment:
This is not a valid kwarg, but it is a valid project option. You can add it to default_options in the project() declaration, or override it per target with `override_options: ['cpp_std=c++17']`